content/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-apis-kafka-spark-streaming-and-machine-learning-part-3.md

```diff
@@ -0,0 +1,295 @@
+
+**What is the count of predicted delay/notdelay by destination?**
+
+```scala
+scala> spark.sql("select dest, pred_dtree, count(pred_dtree) from flight group by dest, pred_dtree order by dest").show
+```
+
+![Delay/NotDelay destination](/img/delay-notdelay-destination.png)
+
+(The complete code, instructions and more example queries are in the github code link at the end.)
+
+**Querying the Data with Apache Drill**
+
+Apache Drill is an open source, low-latency query engine for big data that delivers interactive SQL analytics at petabyte scale. Drill provides a massively parallel processing execution engine, built to perform distributed query processing across the various nodes in a cluster.
+
+![Querying the Data with Apache Drill](/img/querying-data-with-apache-drill.png)
+
+With Drill, you can use SQL to interactively query and join data from files in JSON, Parquet, or CSV format, Hive, and NoSQL stores, including HBase, MapR Database, and Mongo, without defining schemas. MapR provides a [Drill JDBC](https://package.mapr.com/tools/MapR-JDBC/MapR_Drill/) driver that you can use to connect Java applications, BI tools, such as SquirreL and Spotfire, to Drill. Below is a snippit of Java code for querying MapR Database using Drill and JDBC:
+
+![querying MapR Database using Drill and JDBC](/img/querying-mapr-db-using-drill-jdbc.png)
+
+The output for this query **"What is the count of predicted delay/notdelay by origin?"** is shown below:
+
+![output of predicted delay/notdelay by origin](/img/output-delay-notdelay-origin.png)
+
+Below are some example SQL queries using the Drill shell.
+
+**What is the count of predicted delay/notdelay by origin**
+
+![count of predicted delay/notdelay by origin](/img/count-predicted-delay-notdelay-origin.png)
+
+**What is the count of predicted delay/notdelay by origin and dest?**
+
+![What is the count of predicted delay/notdelay by origin and dest?](/img/count-predicted-delay-notdelay-origin-dest.png)
+
+Follow the instructions in the github code README to [add a secondary index to MapR Database](https://docs.datafabric.hpe.com/62/Drill/optimizing_queries_with_indexes.html) and try more queries using the index.
+
+**Querying with the Open JSON API (OJAI)**
+
+Below is a Java example of using the OJAI Query interface to query documents in a MapR Database JSON table:
+
+![OJAI Query interface to query documents in a MapR Database JSON](/img/ojai-query-interface-mapr-db.png)
+
+Partial output for this query to "find predicted late flights for AA" is shown below:
+
+![find predicted late flights for AA](/img/find-predicted-late-flights-aa.png)
+
+Below are some example OJAI queries using the MapR Database shell.
+
+**What are the SFO to DEN flights that were predicted late ?**
+
+```json
+maprdb> find /apps/flights -where '{"$and":[{"$eq":{"pred_dtree":1.0}},{ "$like" : {"_id":"%SFO_DEN%"} }]}' --f _id,pred_dtree
+```
+
+
+![MapR Database find /apps/flights](/img/mapr-db-find.png)
+
+**Summary**
```

Comment:
In the summary and code sections, I would not use bullets, but instead just regular text.

![image](https://user-images.githubusercontent.com/77496261/137183910-ca673704-f9ec-416e-9586-f5eb519d283d.png)

Reply:
done